packages/go/stbernard/graph/transform.go

```diff
@@ -0,0 +1,346 @@
+// Copyright 2025 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+package graph
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log/slog"
+	"strconv"
+	"strings"
+
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/specterops/bloodhound/dawgs"
+	"github.com/specterops/bloodhound/dawgs/drivers/pg"
+	"github.com/specterops/bloodhound/dawgs/graph"
+	"github.com/specterops/bloodhound/dawgs/query"
+	"github.com/specterops/bloodhound/dawgs/util/size"
+	"github.com/specterops/bloodhound/graphschema"
+	"github.com/specterops/bloodhound/graphschema/common"
+	"github.com/specterops/bloodhound/packages/go/stbernard/environment"
+	"github.com/specterops/bloodhound/packages/go/stbernard/workspace"
+	"github.com/specterops/bloodhound/slicesext"
+	"github.com/specterops/bloodhound/src/migrations"
+	"github.com/specterops/bloodhound/src/model"
+	"github.com/specterops/bloodhound/src/services/graphify"
+	"github.com/specterops/bloodhound/src/services/upload"
+)
+
+const (
+	Name  = "graph"
+	Usage = "Ingest test files and transform graph data into an arrows file"
+)
+
+type command struct {
+	env  environment.Environment
+	path string
+	root string
+}
+
+// Create new instance of command to capture given environment
+func Create(env environment.Environment) *command {
+	return &command{
+		env: env,
+	}
+}
+
+// Usage of command
+func (s *command) Usage() string {
+	return Usage
+}
+
+// Name of command
+func (s *command) Name() string {
+	return Name
+}
+
+// Parse command flags
+func (s *command) Parse(cmdIndex int) error {
+	cmd := flag.NewFlagSet(Name, flag.ExitOnError)
+	workspacePaths, err := workspace.FindPaths(s.env)
+	if err != nil {
+		return fmt.Errorf("failed to find workspace paths: %w", err)
+	}
+	s.root = workspacePaths.Root
+
+	path := cmd.String("path", workspacePaths.Root, "destination path for arrows.json file, default is root")
+
+	cmd.Usage = func() {
+		w := flag.CommandLine.Output()
+		fmt.Fprintf(w, "%s\n\nUsage: %s %s [OPTIONS]\n\nOptions:\n", Usage, filepath.Base(os.Args[0]), Name)
+		cmd.PrintDefaults()
+	}
+
+	if err := cmd.Parse(os.Args[cmdIndex+1:]); err != nil {
+		cmd.Usage()
+		return fmt.Errorf("parsing %s command: %w", Name, err)
+	}
+
+	if *path != "" {
+		s.path = *path
+	}
```

Comment:
We can avoid the pointer pain by using `cmd.StringVar` to get our string. Since we're just trying to put the string data into `s.path`, we can remove this code altogether if we use the following for capturing the flag:

```go
cmd.StringVar(&s.path, "path", workspacePaths.Root, "destination path for generic graph file, default is {root}/tmp/graph.json")
```